topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -133,14 +133,14 @@ public void init(Vertx vertx, Context context) {
      * Stop the operator.
      */
     @Override
-    @SuppressWarnings("deprecation") // Uses a deprecated executeBlocking call that should be addressed later. This is tracked in https://github.com/strimzi/strimzi-kafka-operator/issues/9233
     public void stop(Promise<Void> stop) throws Exception {
         this.stopped = true;
         Long timerId = this.timerId;
         if (timerId != null) {
             vertx.cancelTimer(timerId);
         }
-        vertx.executeBlocking(blockingResult -> {
+        Promise<Promise<Void>> blockingResult = Promise.promise();
```

Comment:
I don't think this is correct. `stop` is getting completed based on the `Future` returned by `executeBlocking`, not based on when the `blockingResult` is completed (in fact, no code here is dependent on `blockingResult` completing). By my reading of the code, `stop` will be completed once `longHandler.handle()` has been called once, and all the rest of the code here will run _after_ that. 

I think what's happened here (if this _ever_ worked correctly) is that originally we were just using the `executeBlocking` to block on futures in the appropriate order while shutting down. Over time people have reported blocked thread warnings because (for example) sometimes some of those things just happened to take longer than normal. Those have been 'fixed' by (erroneously) making the shutdown of components async, and forgetting about any ordering between them in the process.

The root of this problem is that vertx provides this primitive for blocking execution, but places this arbitrary soft limitation on how long a blocking operation might take. We used that primitive without any of the methods we were calling offering the guarantee of completing within the soft limit vertx has imposed. We ought to have either said "Sometimes you get the warning from vertx, just because closing your admin client/disconnecting from ZK is slower than expected. It's not a bug", or we should have spun up our own executor for shutting things down with as much blocking as needed without the vertx warnings.

Although there's only ever one `Session` within the lifetime of a `Main.main()` we should still try to fix the shutdown so that components are shutdown in an orderly manner, but it's a bit more work than what we have here.

Reply:
Currently, the ShutdownHook sets a hard cap of 10 seconds to complete Vertx shutdown, which I think it's fine. As you say, we can simply try to stop/close all components sequentially, using our own executor where needed.